lua/lspconfig/configs/gh_actions_ls.lua

```diff
@@ -9,9 +7,21 @@ return {
     -- files. (A nil root_dir and no single_file_support results in the LSP not
     -- attaching.) For details, see #3558
     root_dir = function(filename)
-      return filename:find('/%.(github|forgejo|gitea)/workflows/.+%.ya?ml')
-          and util.root_pattern('.github', '.forgejo', '.gitea')(filename)
-        or nil
+      local dirs_to_check = {
+        '.github/workflows',
+        '.forgejo/workflows',
+        '.gitea/workflows',
+      }
+
+      local dir = vim.fs.dirname(filename)
+      for _, subdir in ipairs(dirs_to_check) do
+        local match = vim.fs.find(subdir, { path = dir, upward = true })[1]
+        if match and vim.fn.isdirectory(match) == 1 and vim.fs.dirname(filename) == match then
+          return match
```

Comment:
What is the intention of this logic? Is it something that needs to be in a `util` function, and potentially find its way into `vim.fs` ?

The commit messages don't give any insight.

Reply:
`vim.fs.relpath()` in Nvim 0.11 may address that. Maybe a code comment here would be helpful